packages/types/src/node.ts

```diff
@@ -64,6 +64,16 @@ export namespace Node {
     reset?(): Promise<void>;
   }
 
+  export interface ILockService {
+    get(path: string): Promise<Lock>;
+    set(path: string, value: Lock): Promise<void>;
```

Comment:
Rename to `tryToGrabLock` or something more indicative of what is does.

Also the error case should be explicit.

Reply:
^^ This.

Copying my suggestion for `ILock` interface from discord here:

```
export interface ILockService {
  acquireLock(timeout: number): Promise<any>;
}
```

The promise is rejected if after `timeout` millis if the lock isn't able to be acquired.

`ILock` would be implemented on objects. If we are using `ILockService` (i.e., if we add a `RedlockService` impl) perhaps the interface should look like this:
```
export interface ILockService {
  acquireLock(callback: (...args: any[]) => any, timeout: number): Promise<any>;
}
```
`callback` would be invoked if the lock was successfully acquired. The promise is rejected if after `timeout` millis if the lock isn't able to be acquired. In this case, `callback` is dropped on the floor.

Reply:
Also for clarity, the `any` retval on the `callback` in this example allows us to return stuff after we get the lock!

We should not return a `Promise<Lock>`. We should return a `Promise<any>` that returns the retval from `callback` and reject the promise if we can't acquire the lock.